### PR TITLE
Drop php 8.0

### PR DESCRIPTION
**Pull request by Xerkus**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   |no
| RFC           | no
| QA            | yes

### Description
- Drop php 8.0 in preparation for phpunit 10.
- Drop obsolete composer conflicts.
- Bump dependencies
- Remove obsolete platform ignore from ci config.

### CODE REVIEW

File: psalm.xml.dist [hunk 1]
```diff
@@ -4,6 +4,9 @@
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xmlns="https://getpsalm.org/schema/config"
     xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
+    findUnusedPsalmSuppress="true"
+    findUnusedBaselineEntry="true"
+    findUnusedCode="false"
```
Comment (gsteel) [inline psalm.xml.dist]: Should this not be `true` so it acts as a signal for uncovered methods and dead code? Issues with PHPUnit data providers can be baselined until the PHPUnit psalm plugin catches up…

Comment (Xerkus) [reply psalm.xml.dist]: Libraries have unused code. ...unless psalm considers test code a usage.

Comment (Ocramius) [reply psalm.xml.dist]: test code is code usage

Comment (Xerkus) [reply psalm.xml.dist]: With enabled unused code psalm reports unused to/from Zend methods. I am inclined to just baseline them.
https://github.com/laminas/laminas-psr7bridge/blob/e8ab789d37a4651e8441d0be69b6c02fd229fcc2/src/Psr7ServerRequest.php#L154-L170

Comment (Ocramius) [reply psalm.xml.dist]: Yes, baselining them is fine :+1:

**Review by gsteel — APPROVED** [review]

**Pull request merged**